src/index.js

```diff
@@ -154,20 +135,36 @@ export default function normalize(obj, key=null) {
     return { entities, normalizations, result };
 }
 
-export function denormalize(key, builder, state, requiredFields) {
+// The validator function should return false if the denormalized entity fails validation, true otherwise.
+// It should also accept three parameters: the denormalized entity, entity key, and key.
+export function denormalize(key, builder, state, validator=null) {
```

Comment:
Can you luno our JavaScript style guide and make this comment a more formal function docstring? I've been trying to add more formal docstrings since it makes it easier to get caught up on code after stepping away from it for awhile